terraform/modules/rds_migration/ecs.tf

```diff
@@ -0,0 +1,128 @@
+        { Name = "DATABASE_URL", valueFrom = var.database_connection_string_arn },
+        { Name = "CF_PASSWORD", valueFrom = aws_secretsmanager_secret.cloudfoundry_password.arn },
+        { Name = "CF_SERVICE", valueFrom = aws_secretsmanager_secret.cloudfoundry_service.arn },
+        { Name = "CF_SPACE", valueFrom = aws_secretsmanager_secret.cloudfoundry_space.arn },
+        { Name = "CF_USERNAME", valueFrom = aws_secretsmanager_secret.cloudfoundry_username.arn }
+      ]
+    }
+  ])
+
+  ephemeral_storage {
+    size_in_gib = var.ecs_task_ephemeral_storage
+  }
+
+  runtime_platform {
+    operating_system_family = "LINUX"
+    cpu_architecture        = "X86_64"
+  }
+
+  lifecycle {
+    # The image will be updated by deployments - irritatingly we can't ignore changes just to the image
+    # If changing other aspects of the container definition we'll need to temporarily not ignore changes
+    # to force the update, ensuring the referenced image is the correct current one
+    ignore_changes = [container_definitions]
+  }
+}
+
+# This ECS task just runs continuously in the background doing nothing. This is so we can exec into it later and check the DB
+resource "aws_ecs_task_definition" "exec_placeholder" {
+  #checkov:skip=CKV_AWS_336:using readonlyRootFilesystem to true breaks the app, as it needs to write to app/tmp/pids for example
+  family                   = "${var.prefix}-exec-placeholder"
+  cpu                      = var.db_migration_task_cpu
+  execution_role_arn       = var.ecs_task_execution_role_arn
+  memory                   = var.db_migration_task_memory #MiB
+  network_mode             = "awsvpc"
+  requires_compatibilities = ["FARGATE"]
+  task_role_arn            = var.ecs_task_role_arn
+
+  container_definitions = jsonencode([
+    {
+      name = "db-migration"
+      #A command that will keep the placeholding task running. Keeps looking for changes in the file (which should always exist) to display
+      command   = ["tail", "-f", "/dev/null"]
+      essential = true
+      image     = var.ecr_repository_url
+
+      logConfiguration = {
+        logDriver = "awslogs"
+        options = {
+          awslogs-group         = aws_cloudwatch_log_group.this.id
+          awslogs-region        = "eu-west-2"
+          awslogs-stream-prefix = var.prefix
+        }
+      }
+
+      secrets = [
+        { Name = "DATABASE_URL", valueFrom = var.database_connection_string_arn }
+      ]
+    }
+  ])
+
+  runtime_platform {
+    operating_system_family = "LINUX"
+    cpu_architecture        = "X86_64"
+  }
+
+  lifecycle {
+    # The image will be updated by deployments - irritatingly we can't ignore changes just to the image
+    # If changing other aspects of the container definition we'll need to temporarily not ignore changes
+    # to force the update, ensuring the referenced image is the correct current one
+    ignore_changes = [container_definitions]
+  }
+}
+
+resource "aws_ecs_service" "db_migration" {

```

Comment:
If I understand your comment correctly, the service is running the `exec_placeholder` task, which I want to keep running (it doesn't do anything) whenever the db migration infra has been created, so that we can exec into it anytime to debug the migration process / check things on the db

Maybe the naming of the service is misleading? Lmk if I've misunderstood and we shout chat to discuss 😃

Reply:
The naming of the service is indeed misleading, since it's not actually running the db migration.

But what I was trying to get at was that I don't think the exec_placeholder task needs to be left running for the entire time that we have the db migration infrastructure in existence (particularly as this will be longer than the time that the actual db migration is running). Unless I've missed something, we should only care about it when we're actually interacting with it right? So I was thinking that we'd spin it up and down as a task when we needed.

Reply:
Ah yes I see what you mean! My thinking was that the infra as a whole shouldn't exist much longer than it needs to, but when it does we should have an exec task ready for any quick debugging/checking we need to do. Nonetheless it's not so hard to spin up so I've updated the ECS to desire 0 `db_migration` tasks (making the naming less confusing) and updated the documentation so we can launch the `exec_placeholder` task as and when required.

Reply:
Can we just skip the service in that case? We can spin up the task on it's own with a cli command (with the `--enable-execute-command` option) rather than needing to change the number on a service.

I'm sort of expecting that infra does end up existing for a while, because we don't want to delay the start of actually migrating if something is going wrong with creating it so I think setting it up is an ahead-of-time thing. (And we probably don't bother to destroy it after creating it for a test)

Reply:
I've removed the ecs service from the implementation, and updated the process documentation so that we can launch the tasks as required using the AWS CLI (without ever needing a service) 😀👍